client/landing/stepper/declarative-flow/internals/components/stepper-loader/index.tsx

```diff
@@ -1,10 +1,30 @@
+import { ProgressBar } from '@wordpress/components';
 import clsx from 'clsx';
-import WordPressLogo from 'calypso/components/wordpress-logo';
 import './style.scss';
 
-const StepperLoader = () => {
-	/* eslint-disable wpcalypso/jsx-classname-namespace */
-	return <WordPressLogo size={ 72 } className={ clsx( 'wpcom-site__logo', 'stepper-loader' ) } />;
+interface StepperLoaderProps {
+	title?: string;
+	subtitle?: React.ReactNode;
+	progress?: number;
+	className?: string;
+}
+
+const StepperLoader: React.FC< StepperLoaderProps > = ( {
+	title,
+	subtitle,
+	progress = -1,
+	className,
+} ) => {
+	return (
+		<div className={ clsx( 'stepper-loader', className ) }>
+			<h1 className="stepper-loader__title">{ title }</h1>
+			<ProgressBar
+				value={ progress >= 0 ? progress * 100 : undefined }
```

Comment:
We should probably make the progress prop range from 0-100 and do the calculation on the consumer side. This way we can just do 
```suggestion
	progress,
	className,
} ) => {
	return (
		<div className={ clsx( 'stepper-loader', className ) }>
			<h1 className="stepper-loader__title">{ title }</h1>
			<ProgressBar
				value={ progress }
```

Reply:
I updated the codebase to use `setProgress` with 0-100 in mind. But to have the indefinite loading (the bar that doesn't follow numbers but goes on indefinitely), we have to not pass a number to value, as

```
const isIndeterminate = ! Number.isFinite( value );
```

Any suggestion?

Reply:
Now we're failing over to -1 when it's undefined, so if remove the -1, it will be undefined.